src/System.IO.Compression/src/System/IO/Compression/GZipStream.cs

```diff
@@ -150,6 +150,18 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        public override ValueTask DisposeAsync()
+        {
+            if (GetType() != typeof(GZipStream))
```

Comment:
For derived types you aren't calling into DeflateStream.DisposeAsync why?

Reply:
Because GZipStream isn't sealed.  This is the unfortunate aspect of adding new virtuals to non-sealed types.  If someone had derived from GZipStream and overridden Dispose, now if someone comes along and calls DisposeAsync instead of Dispose, we still need to make sure we call that type's override.  So since we don't have a good way of just testing whether Dispose is overridden, we just check whether this is a derived type or not, and if it is, we simply delegate to Dispose (via the base implementation); it means we lose out of the async-ness, but it means we don't inadvertently skip the derived type's logic.